include/libc/stddef.h

```diff
@@ -3,9 +3,9 @@
 
 #define NULL ((void*)0)
 
-typedef unsigned long size_t;
+typedef unsigned int size_t;
 
-typedef unsigned int uintptr_t;
+typedef signed int ptrdiff_t;
```

Comment:
I'd rather see `ptrdiff_t` introduced when it will get used

Reading more stuff, it doesn't seem entirely like a "signed size_t", it's specifically for the difference between two pointers inside one array. and size_t is also an array index but not only that

with `s32 data[];` then `&data[5] - &data[3]` evaluates to the `ptrdiff_t` value 2 (5 - 3)

and I don't see that being useful anywhere?

Reply:
I don't see any problem with having it there now. Its not like it hurts anything.

Reply:
To be clear with what I meant in my comment above, I think ptrdiff_t is useless because we shouldn't use it